### ISSUE TITLE
User can not win because "Game Won" window does not appear

### ISSUE THREAD

**Issue by KonstantinosGal** [OSSpk/Minesweeper-Desktop-Game#2]
## Logic Bug

I noticed that when the user has pressed (and has revealed) all cells that are not mines the window showing him as a winner does not appear. As a result, there is no action the user can do but press a mine box and lose the game or exit the game. Therefore he can never win. I found the reason behind this and I created a pull request in order to prevent this problematic situation. 

I ran the code in IntelliJ IDEA 2021.1.3 (Ultimate Edition) x64 on Windows 10.

I would appreciate it if you could confirm this and answer to me.

### User must had won because mines are 10 in total
![Must have won](https://user-images.githubusercontent.com/56201820/131046035-28f326e3-9312-43f3-9ef1-fd1f2eb87eaf.png)

### Instead he lost because he pressed an existing mine (no other choice)
![Lost](https://user-images.githubusercontent.com/56201820/131046195-02f22dcc-1007-41e3-bbc3-30d4ce1f69cd.png)

**Comment by harismuneer** [comment]
Thanks, @KonstantinosGal for the comments. It seems to me that when I initially developed this game, I was able to win it when all non-bomb tiles were explored. Can you verify if this bug happens all the time?

Moreover @AkaruiYami can you verify this bug?

**Comment by AkaruiYami** [comment]
Hello,  @harismuneer.  I'm not sure about what kind of bug you are referring to but upon trying the game several times.. this is what I found:

# The "Game Won" window will only appear when all the tiles are flagged and reveled.
---


## As shown in 2 images below, the "Game Won" does not appear even thou I already revealed all the non-bomb tiles.

![test1](https://user-images.githubusercontent.com/51882840/174954596-ccd97dea-2934-4f88-a041-46561d2577c2.png)

![test2](https://user-images.githubusercontent.com/51882840/174954648-7e710216-cdb6-41db-9318-6b8134fbb22f.png)


## Only after I flag all the bomb, the "Game Won" window will appears:
![test3](https://user-images.githubusercontent.com/51882840/174954675-85842c26-f703-4d54-82f1-248c02ebaa49.png)
![test3-2](https://user-images.githubusercontent.com/51882840/174954683-18d4b582-710d-49ee-9619-bb833aaa5694.png)


## Here I only marked the bomb without revealing the non-bomb tiles and the "Game Won" window still not appears.

![test6](https://user-images.githubusercontent.com/51882840/174955148-3a64bff0-e257-419c-b04d-f824b507864e.png)

## Only after all tiles were revealed and flagged, the "Game Won" window appears.
![test6-2](https://user-images.githubusercontent.com/51882840/174955217-0c616928-57c1-4e9c-b9a4-065ad3af780f.png)

**Comment by KonstantinosGal** [comment]
@harismuneer yes this bug happens to me all the time. The change I have made to the code (in my pull request) solves this problem.
